packages/cli/source/manifest/compiler.ts

```diff
@@ -242,61 +123,67 @@ const compile = async (): Promise<Manifest> => {
         return manifest;
     });
 
-    let app: App | null = null;
-    let queries: {
-        [key: string]: Query;
-    } = {};
-    let resources: {
-        [key: string]: Resource;
-    } = {};
+    let manifestResult = {
+        app: {},
+        queries: {},
+        resources: {},
+    };
+
+    let queries: Query[] = [];
+    let resources: Resource[] = [];
+
     for (const manifest of manifests) {
         for (const key in manifest) {
             switch (key) {
                 case "app": {
-                    if (app) {
+                    if (!lodash.isEmpty(manifestResult.app)) {
                         logDuplicateError("app", manifest.file);
                     }
-                    app = parseApp(manifest.app, manifest.file);
+
+                    const { error, value } = appSchema.validate(manifest.app, {
+                        stripUnknown: true,
+                    });
+
+                    if (error) {
+                        logSemanticError(error.message);
+                        break;
+                    }
+                    manifestResult.app = value;
                     break;
                 }
 
                 case "queries": {
-                    queries = {
-                        ...queries,
-                        ...parseQueries(manifest.queries, manifest.file),
-                    };
+                    lodash.merge(
+                        queries,
+                        validateQueries(manifest.queries, manifest.file),
+                    );
```

Comment:
The logic to find duplicates won't work across manifests. It'll work only within the same manifest. The same issue exists with the other entities, too.